src/server.js

```diff
@@ -155,11 +166,13 @@ Server.prototype.receiveEdit = function(connection, editMessage, sendToClient){
         if(!isEmpty(edit.diff)){
           clientDoc.shadow.localVersion++;
         }
+
+        // notify about patch
+        this.emit('patch', { clientDoc, connection, edit, editMessage });
```

Comment:
Maybe we should copy pass a deep copy of the objects since we don't know if the application will mutate those references.